doc/Code/LocalizationFactor.cpp

```diff
@@ -2,11 +2,14 @@ class UnaryFactor: public NoiseModelFactor1<Pose2> {
   double mx_, my_; ///< X and Y measurements
   
 public:
+
+  // Provide access to the Matrix& version of evaluateError:
+  using gtsam::NoiseModelFactor1<Pose2>::evaluateError;
```

Comment:
Why is this necessary?

Reply:
This is to help compiler find the right overload of `evaluateError`. Since a function with the same name is overridden in the derived class (the pointer version), the compiler function name lookup logic just stops in the derived class and complains that it could not find the right overload. By declaring that we use other signatures from the parent class, we are able to access the matrix reference style of `evaluateError`, defined in the parent class, as well.